### PR TITLE
feat(dashboards): add datasets field to DashboardInfo aspect

### PR DESCRIPTION
**Pull request by Masterchen09**
This PR adds an optional datasets field/relationship to the DashboardInfo aspect and a corresponding tab in the UI of the dashboard. To preseve the current behavior the tab with the consumed datasets is only shown if there are any datasets which are directly consumed by the dashboard (the charts tab is hidden if there are only datasets and no charts for the dashboard available). The tab with the consumed datasets shows only the datasets which are directly consumed by the dashboard, not the datasets which are consumed by a chart which is used in the dashboard (this is another reason why the tab with the datasets should not be shown if there are no datasets which are directly consumed by the dashboard).

Why do we need this extra field/relationship? In some dashboarding tools there is no dedicated object which could represent a chart or it is technically not possible (or way too complex in terms of effort) to extract such information from the source system (or manually maintain it). For instance in SAP Lumira or SAP Analytics Cloud there are no dedicated information about a chart available, only information about the data source (i.e. an SAP BW Query) or the model in SAP Analytics Cloud are available. Therefore for each dashboard one or many unnecessary "dummy" charts have to be created without any added value (and of course these dummy charts are visible in the UI which is not really nice when there are also dashboarding tools which are supporting the extraction of chart enities).

Please let me know if something is missing in the implementation as this is my first contribution which adds something to an (existing) aspect (and the UI). I have tried to understand and follow the implementation of the charts and some local testings are looking good, but who knows if I haven't forgotten something. 😁

 
## Checklist
- [x] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [x] Links to related issues (if applicable)
- [x] Tests for the changes have been added/updated (if applicable)
- [x] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [x] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-models/src/main/pegasus/com/linkedin/dashboard/DashboardInfo.pdl [hunk 1]
```diff
@@ -47,6 +48,18 @@ record DashboardInfo includes CustomProperties, ExternalReference {
...
+   * Datasets consumed by a dashboard
+   */
+  @Relationship = {
+    "/*": {
+      "name": "Consumes",
+      "entityTypes": [ "dataset" ],
+      "isLineage": true
+    }
+  }
+  datasets: array[DatasetUrn] = [ ]
```
Comment (chriscollins3456) [inline metadata-models/src/main/pegasus/com/linkedin/dashboard/DashboardInfo.pdl:61]: one minor comment: we're actually trying to move away from using specific urns in general like this, instead could you use `array[Urn]`?

**Review by chriscollins3456 — COMMENTED** [review]
looks good! one minor comment here

File: metadata-models/src/main/pegasus/com/linkedin/dashboard/DashboardInfo.pdl (hunk 1 above)
Comment (Masterchen09) [reply metadata-models/src/main/pegasus/com/linkedin/dashboard/DashboardInfo.pdl]: Is done. 😊

https://github.com/datahub-project/datahub/pull/5188/files#diff-6ae54ee7e3babcd8ac0b99538e8de6725f9fd82c0a119e517c20b2a23575e303R61

Comment (jjoyce0510) [reply metadata-models/src/main/pegasus/com/linkedin/dashboard/DashboardInfo.pdl]: Thanks!

**Comment by jjoyce0510** [comment]
@Masterchen09 Looks like there's just one remaining conflict - otherwise LGTM. If we can fix that up we are good to go.

**Comment by Masterchen09** [comment]
@jjoyce0510 I have rebased the branch and the remaining merge conflict in tableau_mces_golden.json should be solved. :-)

**Pull request merged**